lib/mirage/mirage.mli

```diff
@@ -764,6 +764,114 @@ val default_argv : argv impl
 val no_argv : argv impl
 (** [no_argv] Disable command line parsing and set argv to [|""|]. *)
 
+(** {2 Git client configuration} *)
+
+(** The user is able to initiate a connection with a Git repository. However,
+    there are many ways to communicate with a Git repository:
+
+    - TCP/IP
+    - HTTP
+    - HTTP + TLS
+    - SSH
+
+    The devices defined below define these composable ways. The
+    [git_client impl] returned from them can be passed to Git or Irmin in order
+    to be able to {i fetch} and {i push} from/into a Git repository.
+
+    The user is able to restrict or enlarge protocol possibilities needed for
+    its application. For instance, the user is able to restrict only the SSH
+    connection to communicate with a Git repository or the user can handle
+    TCP/IP and SSH as possible protocols to communicate with a peer.
+
+    For instance, a device which is able to communicate {i via} TCP/IP and SSH
+    can be implemented like:
+
+    {[
+      let git_client =
+        let dns = git_happy_eyeballs stackv4v6 in
```

Comment:
Should we call this device `dns`? Why does `git_happy_eyeballs` is related to `git` at all?

Reply:
from a brief look into "Git_mirage_happy_eyeballs", this seems to depend and use "mimic". I think the code should be part of mimic. `happy-eyeballs{-mirage}` is pretty concise about its footprint, and useful in other settings than mimic. The scope and API is basically https://datatracker.ietf.org/doc/html/rfc8305.

Reply:
ok, so would renaming `type git_client_with_dns` into `type happy_eyeball` make sense? We could move the code later on (I'm just focusing on the API surfaced in mirage.mli here)

Reply:
you don't have to do any changes in `mimic` yet, let's try to have a nice API in mirage.mli first (and merge this PR). I think we are getting closer now :-)